gui/gui.py

```diff
@@ -0,0 +1,136 @@
+from pygame.font import Font
+from config.settings.general_config import Gui_font, Colors, Window
+from config.settings.towers import *
+from game_objects.game_object import GameObject
+from graphics_manager.graphics_manager import GraphicsManager
+
+from pygame import draw, display, Surface
+from pygame.sprite import RenderUpdates, Sprite
+from pygame.transform import scale
+
+
+class Gui():
+    def __init__(self, gfx_manager:GraphicsManager) -> None:
+        self.graphics_manager = gfx_manager
+        self.position  = Window.GUI_POS
+        self.font_size = Gui_font.size
+        self.path = Gui_font.path
+        self.color = Colors.MENU_TEXT
+        self.background_color = Colors.MENU_BG
+
+        self.background = None
+        self.changed_rects = []
+
+        self.font = Font(self.path, self.font_size)
+        self.gui_scale = Window.GUI_SCALE
+
+        self.gui_height = Window.GUI_HEIGHT
+        self.gui_width = Window.GAME_WIDTH
+        self.center_for_text = (self.gui_height-self.font_size)//2
+        # Tohle bude chtit algoritmus na automaticke zarovnani
+        self.lives_pos = (self.gui_width-100, self.center_for_text)
+        self.coins_pos = (self.gui_width-230, self.center_for_text)
+        self.wave_pos = (self.gui_width-320, self.center_for_text)
+
+        self.lives = RenderUpdates()
+        self.coins = RenderUpdates()
+        self.old_coins_text = None
+        self.wave = RenderUpdates()
+        self.pause = RenderUpdates()
+
+        self.towers_pos = []
+        self.tower_pos_types = []
+
+        self.tower_cards = RenderUpdates()
+
+
+    def create_gui(self, lives:int, coins:int, wave:int, textures:dict):
+        """ Creates the gui"""
+        self.draw_background()
+        self.show_lives(lives)
+        self.show_coins(coins)
+        self.show_wave(wave)
+        self.show_towers(textures)
+
+
+    def draw_background(self):
+        """ Draws the background of the gui"""
+        #draw.rect(screen, self.background_color, (self.position[0], self.position[1], Window.WIDTH, Window.GUI_HEIGHT))
+        self.background = Surface((Window.GUI_WIDTH, Window.GUI_HEIGHT))
+        self.background.fill(self.background_color)
+
+        background_object = GameObject(0, 0, Window.GUI_WIDTH, Window.GUI_HEIGHT, self.background)
+        self.graphics_manager.draw_object(background_object, False)
+
+
+    def create_towers_grid(self):
+        """ Creates the grid of tower cards"""
+        for tower, tower_type in enumerate(tower_types):
+            position = (Window.CARD_RECT_WIDTH+tower*Window.PIXEL_SIZE*self.gui_scale+tower*Window.CARD_RECT_WIDTH, (self.gui_height-Window.PIXEL_SIZE*self.gui_scale)//2)
+            self.towers_pos.append(position)
```

Comment:
Ty máš seznam pozic věží a pak seznam dvojic (pozice, typ věže). Hmm... Neříká se téhle struktuře třeba SLOVNÍK?

Reply:
slovník je taková ta kniha jak jsou v ní slova co to má společnýho??